treeherder/auth/backends.py

```diff
@@ -0,0 +1,50 @@
+import logging
+from hashlib import sha1
+
+from django.contrib.auth.models import User
+from django.db.models import Q
+from taskcluster.sync import Auth
+
+logger = logging.getLogger(__name__)
+
+
+class TaskclusterAuthBackend(object):
+
+    def authenticate(self, authorization=None, host=None, port=None):
+        user = None
+
+        if authorization:
+            tc_auth = Auth()
+            result = tc_auth.authenticateHawk({
+                "authorization": authorization,
+                "host": host,
+                "port": port,
+                "resource": "/",
+                "method": "get",
+            })
+
+            if result["status"] == "auth-success":
+                username = result["clientId"][-30:]
+                email = result["clientId"].split("/")[1]
```

Comment:
This is assuming a clientId of the form `<something>/<email>` which happens to match the temporary credentials that taskcluster-login hands out, but not all clientIds have that form; see the full list of permanent credentials (which can be shared just as easiliy) at https://tools.taskcluster.net/auth/clients/

ClientIds can be used as visual feedback ("Hello, dmitchell@mozilla.com!"), but should not be used for access control ("Our records show that dmitchell@mozilla.com has permission to ..").  It's a little hard to tell which is the case here?